main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -464,15 +468,147 @@ DataTableView.prototype._renderDataTables = function(table, tableHeader) {
+      } else {
+        var tr = table.insertRow(table.rows.length);
+      }
+      if (theProject.rowModel.mode == "row-based" || "j" in row) {
+        even = !even;
+      }
+      renderRow(tr, r, row, even);
     }
-    renderRow(tr, r, row, even);
   }
+  loadRows();
+
+  var total = 0, row;
+  for (var i = 1; i < table.rows.length; i++) {
+      row = table.rows[i];
+      total += row.offsetHeight;
+  }
+  this._sizeRowFirst = total / this._pageSize;
+  this._sizeRowsTotal = this._sizeRowFirst * theProject.rowModel.filtered;
+  this._sizeSinglePage = this._sizeRowFirst * this._pageSize;
+  document.querySelector('.data-table tbody').insertRow(0).setAttribute('class', 'first-row');
+  this._headerTop = $('thead').offset().top + $('thead').height();
+  this._pageStart = 0;
+  this._totalSize = this._pageSize;
+  this._adjustNextSetClasses();
+
+  var flag = 0;
+  $(table.parentNode.parentNode).bind('scroll', function(evt) {
+    self._downwardDirection = self._scrollTop < $(this).scrollTop();
+
+    try {
+      var element = document.querySelectorAll('.load-next-set');
+      element = element[element.length - 1];
+      var position = element.getBoundingClientRect();
+      var element2 = document.querySelector('.load-next-set');
+      var position2 = element2.getBoundingClientRect();
+    } catch (err) {
+      var position = {top: undefined, bottom: undefined};
+      var position2 = {top: undefined, bottom: undefined};
+    }
+    var lastElement = document.querySelector('.last-row');
+    var positionLastElement = lastElement.getBoundingClientRect();
+    var firstElement = document.querySelector('.first-row');
+    var positionFirstElement = firstElement.getBoundingClientRect();
+    
+    if(self._downwardDirection && !flag) {
+      if((position.top >= 0 && position.bottom <= window.innerHeight) || 
+        (positionLastElement.top < window.innerHeight && positionLastElement.top > 0)) {
+        flag = true;
+        self._onBottomTable(self._scrollTop, table, this, evt);
+      }
+    }
+
+    if(!self._downwardDirection && !flag) {
+      if(position2.top >= 0 && position2.bottom <= window.innerHeight || 
+        (positionFirstElement.bottom > 0 && positionFirstElement.bottom < window.innerHeight)) {
+        flag = true;
+        self._onTopTable(self._scrollTop, table, this, evt);
+      }
+    }
+
+    clearTimeout($.data(this, 'scrollTimer'));
+    $.data(this, 'scrollTimer', setTimeout(function() {
+      if(positionLastElement.top <= 0 && positionLastElement.bottom >= 0) {
+        var goto = self.getPageNumberSrcolling(self._scrollTop);
+        self._onChangeGotoScrolling(self._scrollTop, goto, table);
+      }
+
+      if(positionFirstElement.top < self._headerTop + 1 && positionFirstElement.bottom >= window.innerHeight) {
+        var goto = self.getPageNumberSrcolling(self._scrollTop);
+        self._onChangeGotoScrolling(self._scrollTop, goto, table);
+      }
+      flag = 0;
+    }, 250));
```

Comment:
This feels too long to me. If I'm scrolling with the scroll wheel, the screen basically goes blank until I stop scrolling. If I'm using the scroll wheel, I want to be seeing at least a few updates a second. I'm not sure if reducing this to 100 msec would help or if a different heuristic is needed (e.g. throttling the update rate rather than shutting off updates entirely if the scroll rate is too high).

Reply:
The reason I didn't go for throttling is because the amount of time required to render rows limits the fps to a great extent, even if we allow something like 60 fps (the most common rate followed).